src/watch.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{cmd, util::Fallible};
+
+use clap::ArgMatches;
+use notify::{DebouncedEvent, Watcher};
+use std::time::Instant;
+use std::{collections::HashSet, env, path::PathBuf, sync::mpsc::channel, time::Duration};
+
+const COLLECT_EVENTS_MILLIS: u64 = 500;
+
+#[derive(PartialEq)]
+pub enum ChangeType {
+    Sources,
+    Code,
+}
+
+pub fn watch(
+    args: ArgMatches,
+    watch_sources: HashSet<PathBuf>,
```

Comment:
Do you really need to collect this into a hash table?

Reply:
No, not really. So should I just use `Iterator` (or `&mut Iterator` in `trigger_on_change`)?

Reply:
`impl Iterator<Item=PathBuf>` or something like it should work.